build.gradle

```diff
@@ -138,7 +140,59 @@ task checkPlatformConfig {
     }
 }
 
-task copyBaseLibs(dependsOn:configurations.corelibs) {
+task copyEngineLibs() {
+    description = "Copy libraries from engine directory before configuring gradle"
+
+    doFirst {
+        // Collect dependencies in corelibs, optlibs, testdatalibs...
+        def allDependencies = project.configurations.collect { conf -> conf.incoming.dependencies }.flatten()
+        allDependencies.each { file -> copyEngineJarsToLib(file) }
+    }
+}
+
+def copyEngineJarsToLib(file, extraJars=["nifty"]) {
```

Comment:
I still don't gather why you try to copy the files from the engine folder, where the idea is to use gradles dependency resolution?

Also why do you depend on nifty-gui through jitpack while we get it transitively through https://github.com/jMonkeyEngine/sdk/blob/36d11b6f712539693e34bbc79009121c11aa1d53/build.gradle#L99 ?

Reply:
@MeFisto94 
Because, in my understanding, the `buildSdk` task need to have dependency jars but there are no process to download or copy these jar files. Probably since that reason,  on my local environment after executing `buildSdk` will cause gradle error "Could not find org.jmonkeyengine:jme3-*".

I'm afraid current build script don't resolve dependency jars (jme3-*.jar) correctly. `build_engine.sh` generates many jme3-*.jar files, but it don't copy files into `/lib` directory or maven local repository.
So, I modified build.gradle script. Is there any misunderstanding for my side ?

Reply:
That's definitely a misunderstanding, because CI works (https://travis-ci.org/github/jMonkeyEngine/sdk/jobs/724436054).
If that error was about a specific jme3-jogl, that's fixed in 56cdc94e55a18b4b05bf8e77467db08802c6d491, essentially: Your engine folder needs to be on 3.3 (checkout or delete it and let the scripts work).

Due to https://github.com/jMonkeyEngine/sdk/blob/36d11b6f712539693e34bbc79009121c11aa1d53/settings.gradle#L2, engine dependencies should be there (unless jogl, which got deleted by the engine)

https://github.com/jMonkeyEngine/sdk/blob/36d11b6f712539693e34bbc79009121c11aa1d53/build.gradle#L167 Is the part which is copying them into a netbeans module, from which it is then available.

So all in all, it should work. Sorry for being not clear enough about this in the beginning, so you had that extra work